framework/meta/src/cmd/standalone/upgrade/upgrade_selector.rs

```diff
@@ -69,41 +78,32 @@ fn upgrade_function_selector(dir: &RelevantDirectory) {
         print_upgrading(dir);
     }
 
-    match dir.upgrade_in_progress {
-        Some((_, "0.31.0")) => {
-            upgrade_to_31_0(dir);
-        },
-        Some((_, "0.32.0")) => {
-            upgrade_to_32_0(dir);
-        },
-        Some((_, "0.39.0")) => {
-            upgrade_to_39_0(dir);
-        },
-        Some((_, "0.45.0")) => {
-            upgrade_to_45_0(dir);
-        },
-        Some((from_version, to_version)) => {
-            version_bump_in_cargo_toml(&dir.path, from_version, to_version);
-        },
-        None => {},
+    if let Some((from_version, to_version)) = dir.upgrade_in_progress {
+        match to_version.version.to_string().as_str() {
+            "0.31.0" => upgrade_to_31_0(dir),
```

Comment:
Great place to put our new macro to use, and match on the FrameworkVersion directly.